_rules/table-header-cell-has-assigned-cells-d0f69e.md

```diff
@@ -27,20 +27,19 @@ acknowledgments:
 
 ## Applicability
 
-This rule applies to any HTML element with the [semantic role][] of [rowheader][] or [columnheader][] for which all of the following is true:
+This rule applies to any HTML element with the [semantic role][] of [rowheader][] or [columnheader][] within a [table][] or [grid][] element, where the [table][] or [grid][] element is [visible][].
 
-- the element is [visible][]; and
-- the element is [included in the accessibility tree][]; and
-- the element has at least one ancestor in the [flat tree][] with a [semantic role][] of either [table][] or [grid][]; and
-- the element's closest ancestor in the [flat tree][] with a [semantic role][] of either [table][] or [grid][] is [included in the accessibility tree][].
+## Expectation 1
 
-## Expectation
+Each target element is [assigned][] to at least one element with a [semantic role][] of [cell][]. The test target and the [cell][] are within an element with the [semantic role][] of [table][] or [grid][].
```

Comment:
The second sentence looks like a different expectation.
Also, since the target is part of the table, maybe
```suggestion
Each target element is [assigned][] to at least one element with a [semantic role][] of [cell][]. which is within the same element with the [semantic role][] of [table][] or [grid][].
```
(+ same caveat about "within")

Reply:
Ah no, I get it… (reading the next expectation…)
So more like
```suggestion
For each test target, either it is not part of a [table][]; or it is [assigned][] to at least one element with a [semantic role][] of [cell][] which is within the sameelement with the [semantic role][] of [table][].
```

Reply:
Not sure why "either it is not part of a [table][]"? Elements with the semantic role of `rowheader` or `columnheader` have required context roles that eventually end on an element with the semantic role of either `grid` or `table`.

Reply:
You need the "either" bit because of the table/grid duality:
```html
<table role="grid">
<tr><th role=columnheader>Foo</th></tr>
<tr><td role="gridcell">Bar</td></tr>
</table>
```
Now, this `th` is not assigned to any element with a role of `cell` and thus fails the expectation. Yet, it is good because it is inside a `grid`.

When there are several expectations, a target needs to pass **all** of them in order to pass the rule.

Reply:
I think that I've addressed that with the current format of the expectations section. The second expectation should account for the `table`/`grid` duality.